ibis/backends/dask/execution/aggregations.py

```diff
@@ -0,0 +1,143 @@
+
+    if op.sort_keys:
+        raise NotImplementedError(
+            'sorting on aggregations not yet implemented'
+        )
+
+    predicates = op.predicates
+    if predicates:
+        predicate = functools.reduce(
+            operator.and_,
+            (
+                execute(p, scope=scope, timecontext=timecontext, **kwargs)
+                for p in predicates
+            ),
+        )
+        data = data.loc[predicate]
+
+    columns = {}
+
+    if op.by:
+        grouping_key_pairs = list(
+            zip(op.by, map(operator.methodcaller('op'), op.by))
+        )
+        grouping_keys = [
+            by_op.name
+            if isinstance(by_op, ops.TableColumn)
+            else execute(
+                by, scope=scope, timecontext=timecontext, **kwargs
+            ).rename(by.get_name())
+            for by, by_op in grouping_key_pairs
+        ]
+        columns.update(
+            (by_op.name, by.get_name())
+            for by, by_op in grouping_key_pairs
+            if hasattr(by_op, 'name')
+        )
+        source = data.groupby(grouping_keys)
+    else:
+        source = data
+
+    scope = scope.merge_scope(Scope({op.table.op(): source}, timecontext))
+
+    pieces = []
+    for metric in op.metrics:
+        piece = execute(metric, scope=scope, timecontext=timecontext, **kwargs)
+        piece.name = metric.get_name()
+        pieces.append(piece)
+
+    result = dd.concat(pieces, axis=1)
+
+    # If grouping, need a reset to get the grouping key back as a column
+    if op.by:
+        result = result.reset_index()
+
+    result.columns = [columns.get(c, c) for c in result.columns]
+
+    if op.having:
+        # .having(...) is only accessible on groupby, so this should never
+        # raise
+        if not op.by:
+            raise ValueError(
+                'Filtering out aggregation values is not allowed without at '
+                'least one grouping key'
+            )
+
+        # TODO(phillipc): Don't recompute identical subexpressions
+        predicate = functools.reduce(
+            operator.and_,
+            (
+                execute(having, scope=scope, timecontext=timecontext, **kwargs)
+                for having in op.having
+            ),
+        )
+        assert len(predicate) == len(
+            result
+        ), 'length of predicate does not match length of DataFrame'
+        result = result.loc[predicate.values]
+    return result
+
+
+# TODO - aggregations - #2553
+# @execute_node.register((ops.Any, ops.All), (dd.Series, SeriesGroupBy))
+# def execute_any_all_series(op, data, aggcontext=None, **kwargs):
+#     if isinstance(aggcontext, (agg_ctx.Summarize, agg_ctx.Transform)):
+#         result = aggcontext.agg(data, type(op).__name__.lower())
+#     else:
+#         result = aggcontext.agg(
+#             data, lambda data: getattr(data, type(op).__name__.lower())()
+#         )
+#     return result
+
+# TODO - aggregations - #2553
```

Comment:
I'd rather leave aggregations as is with specific tests xfailed. Some functionality ends up going through the aggregations handler without encountering paths that dask doesn't support. For example, out of just dask tests, the following would fail if the current handler just errored out (they pass cleanly otherwise). 

```
FAILED ibis/backends/dask/tests/execution/test_arrays.py::test_array_collect - NotImplementedError: Aggregations are still todo
FAILED ibis/backends/dask/tests/execution/test_operations.py::test_group_by_with_having - NotImplementedError: Aggregations are still todo
FAILED ibis/backends/dask/tests/execution/test_operations.py::test_group_by_rename_key - NotImplementedError: Aggregations are still todo
FAILED ibis/backends/dask/tests/execution/test_operations.py::test_group_concat - NotImplementedError: Aggregations are still todo
FAILED ibis/backends/dask/tests/execution/test_operations.py::test_value_counts - NotImplementedError: Aggregations are still todo
FAILED ibis/backends/dask/tests/execution/test_operations.py::test_group_by_multiple_keys - NotImplementedError: Aggregations are still todo
FAILED ibis/backends/dask/tests/execution/test_operations.py::test_mutate_after_group_by - NotImplementedError: Aggregations are still todo
FAILED ibis/backends/dask/tests/execution/test_structs.py::test_struct_field_series_group_by_key - NotImplementedError: Aggregations are still todo
```